integration_test/tests/power_bi_tests.rs

```diff
@@ -531,4 +531,185 @@ mod integration {
+        let mut stmt: Handle = null_mut();
+        let str_len_ptr = &mut 0;
+        let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+            let column_count_ptr = &mut 0;
+            let mut table_view: Vec<u16> = "TABLE,VIEW".encode_utf16().collect();
+            table_view.push(0);
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLTablesW(
+                    stmt as HStmt,
+                    null_mut(),
+                    0,
+                    null_mut(),
+                    0,
+                    null_mut(),
+                    0,
+                    table_view.as_ptr(),
+                    10
+                ),
+                "{}",
+                get_sql_diagnostics(HandleType::Env, env_handle as Handle)
+            );
+
+            // SQLGetFunctions is not available through odbc_sys
+            // SQLGetFunctions(SQL_API_SQLFETCHSCROLL)
+
+            test_get_info!(
+                conn_handle,
+                InfoType::GetDataExtensions,
+                2,
+                DataType::USmallInt
+            );
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLNumResultCols(stmt as HStmt, column_count_ptr)
+            );
+            let numeric_attribute_ptr = &mut 0;
+            const FIELD_IDS: [Desc; 7] = [
+                Desc::ConciseType,
+                Desc::Unsigned,
+                Desc::Name,
+                Desc::Nullable,
+                Desc::TypeName,
+                Desc::Length,
+                Desc::Scale,
+            ];
+            let expected_col_attr_string_lengths = [
+                &[0, 0, 18, 18, 12, 12, 12],
+                &[12, 12, 22, 22, 12, 12, 12],
+                &[12, 12, 20, 20, 12, 12, 12],
+                &[12, 12, 20, 20, 12, 12, 12],
+                &[12, 12, 14, 14, 12, 12, 12],
+            ];
+            for col_num in 0..*column_count_ptr {
+                let mut field_num: usize = 0;
+                FIELD_IDS.iter().for_each(|field_type| {
+                    assert_eq!(
+                        SqlReturn::SUCCESS,
+                        SQLColAttributeW(
+                            stmt as HStmt,
+                            (col_num + 1) as u16,
+                            *field_type,
+                            output_buffer as Pointer,
+                            BUFFER_LENGTH,
+                            str_len_ptr,
+                            numeric_attribute_ptr,
+                        ),
+                        "{}",
+                        get_sql_diagnostics(HandleType::Stmt, stmt as Handle)
+                    );
+                    assert_eq!(
```

Comment:
There is already a good coverage for SQLColAttribute tests via the test harness and tests in queries.yaml. I think that you can skip this validation. 
The PowerBi flows are here more to make sure that we can calls the same functions, with similar input arguments, and this way at least know things are not broken. The other issue I see is that everytime we add a new database or more tables, the `expected_col_attr_string_lengths` array will need to be amended.

Reply:
Sounds good, I added this check based on the last PR comments regarding checking the str_len_ptr length.  You are right in that this is already checked and would require this test to also be updated when the data set changes.  I have removed these validations.